src/networks/invertible_network_conditional_hint_multiscale.jl

```diff
@@ -257,10 +298,10 @@ function jacobian(ΔX, ΔY, Δθ::Array{Parameter, 1}, X, Y, CH::NetworkMultiSca
     GNΔθ = Array{Parameter, 1}(undef, 0)
     idxblk = 0
     for i=1:CH.L
-        X = wavelet_squeeze(X)
-        ΔX = wavelet_squeeze(ΔX)
-        Y = wavelet_squeeze(Y)
-        ΔY = wavelet_squeeze(ΔY)
+        X  = general_squeeze(X; squeeze_type=CH.squeeze_type)
+        ΔX = general_squeeze(ΔX; squeeze_type=CH.squeeze_type)
+        Y  = general_squeeze(Y; squeeze_type=CH.squeeze_type)
+        ΔY = general_squeeze(ΔY; squeeze_type=CH.squeeze_type)
```

Comment:
Fo everywhere, rather than this not so nice "general_squeeze" what you could do is have a `CH.squeezer` at setup that is basically obtained from either the if/esle or a Dict `Dict(name=> squezze_func)` thaty you just read at setup `(CH.squeezer =  _squeezers[name])`.

Reply:
I couldnt figure out how to do this cleanly considering that different squeezes have different arguments. So I made the squeeze a structure that defines its forward and inverse. That is then an option to the user just like the activationfunction is an option when constructing a network.